### DISCUSSION TITLE
pool cannot be imported read-write

### DISCUSSION THREAD

**Discussion by gardar** [zbm-dev/zfsbootmenu#677] [Questions] [answered]
What are the requirements for importing pools as read-write? The documentation repeatedly states:

> The operation will fail gracefully if the pool cannot be set to read/write.

However, there is no clear explanation on how to ensure the pool can be imported as read/write or why it might fail.

This is my first time attempting system recovery, and when trying a r/w chroot, I'm receiving the error:
`Operation prohibited, pool cannot be imported read-write.`

**Comment by ahesford** [comment]
If ZBM refuses to import your look read/write, then either there is a ZFS version incompatibility or the pool is degraded. If the problem is a ZFS version incompatibility, you are stuck. You'll need to build a new image with a newer ZFS, or else use another recovery tool. If the problem is a degraded pool, we refuse to automatically import read/write, but you can drop to the ZBM recovery shell and import the pool manually. After it's reimported, the rest of the ZBM UI should function as expected.

**Reply by gardar** [reply]
Thanks for the hint, it seems to stem from zbm thinking there is some version incompatibility present.

Upon entering the emergency shell I see:
```
ZFSBootMenu: unsupported property: unsupported@com.delphix:redaction_list_spill
```

But if I look at the [feature flags list](https://openzfs.github.io/openzfs-docs/Basic%20Concepts/Feature%20Flags.html) it looks like that flag should be supported.
`zfs --version` in the emergency shell I get:

```
zfs-2.2.4-1
zfs-kmod-2.2.4-1
```

And I can do `zpool export` and `zpool import` just fine in the emergency shell, which makes my pool `read-write`.

**Reply by classabbyamp** [reply, marked as answer]
that flag is only supported on unreleased versions of zfs:

![image](https://github.com/user-attachments/assets/3beaaa8d-573b-40ec-901a-58eac6528ed4)

you will need to build a copy of ZBM with the same version
